woocommerce/woocommerce-android
# IAP: hit the backend if makepurchase called when its been purchased already

Closes: #7627


### Description

The PR:
* Adds "loading" to the showcase so the purchasing progress can be visible
* In cases when we are trying to purchase an IAP that has been purchased already and acknowledged then we just return success, if not acknowledged then we do a network call which should do the acknowledging call to the GP
* Because "purchasing" logic is way too big I extracted it into a separated class 

### Testing instructions

* Make a purchase
* Make it again and notice that the network is done (look at the logs)

### Images/gif


https://user-images.githubusercontent.com/4923871/200300040-717b9748-6a03-4b5b-9687-66107e89a1a0.mp4
